minerl/Malmo/Minecraft/src/main/java/com/microsoft/Malmo/Utils/JSONWorldDataHelper.java

```diff
@@ -112,33 +116,63 @@ public GridDimensions(int xMargin, int zMargin) {
+    public static void buildBaseMinecraftStats(JsonObject json, EntityPlayerSP player)
     {
         if(Minecraft.getMinecraft().isIntegratedServerRunning()){
 
-            StatisticsManagerServer sfw = Minecraft.getMinecraft().getIntegratedServer().getPlayerList().getPlayerStatsFile(player);
+            StatisticsManagerServer sfw = Objects.requireNonNull(Minecraft.getMinecraft().getIntegratedServer()).getPlayerList().getPlayerStatsFile(player);
 
-            json.addProperty("distance_travelled",
+            json.addProperty("distance_travelled_cm",
                 sfw.readStat(StatList.WALK_ONE_CM)
+                + sfw.readStat(StatList.CROUCH_ONE_CM)
+                + sfw.readStat(StatList.SPRINT_ONE_CM)
                 + sfw.readStat(StatList.SWIM_ONE_CM)
-                + sfw.readStat(StatList.DIVE_ONE_CM)
                 + sfw.readStat(StatList.FALL_ONE_CM)
-                ); // TODO: there are many other ways of moving!
-            json.addProperty("time_alive", sfw.readStat(StatList.TIME_SINCE_DEATH));
-            json.addProperty("mobs_killed", sfw.readStat(StatList.MOB_KILLS));
-            json.addProperty("players_killed", sfw.readStat(StatList.PLAYER_KILLS));
-            json.addProperty("damage_taken", sfw.readStat(StatList.DAMAGE_TAKEN));
-            json.addProperty("damage_dealt", sfw.readStat(StatList.DAMAGE_DEALT));
-        }
-
-
+                + sfw.readStat(StatList.CLIMB_ONE_CM)
+                + sfw.readStat(StatList.FLY_ONE_CM)
+                + sfw.readStat(StatList.DIVE_ONE_CM)
+                + sfw.readStat(StatList.MINECART_ONE_CM)
+                + sfw.readStat(StatList.BOAT_ONE_CM)
+                + sfw.readStat(StatList.PIG_ONE_CM)
+                + sfw.readStat(StatList.HORSE_ONE_CM)
+                + sfw.readStat(StatList.AVIATE_ONE_CM)
+                );
 
-        /* Other potential reinforcement signals that may be worth researching:
-        json.addProperty("BlocksDestroyed", sfw.readStat((StatBase)StatList.objectBreakStats) - but objectBreakStats is an array of 32000 StatBase objects - indexed by block type.);
-        json.addProperty("Blocked", ev.player.isMovementBlocked()) - but isMovementBlocker() is a protected method (can get round this with reflection)
-        */
+            for(StatBase stat : StatList.ALL_STATS) {
+                // For MineRL, split over . and convert all camelCase to snake_case
+                String[] stat_fields = stat.statId.split("\\.");
+                JsonObject head = json;
+                for (String unformatted_token : stat_fields) {
+                    String token = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, unformatted_token);
+                    // Last element is a leaf
+                    if (unformatted_token.equals(stat_fields[stat_fields.length - 1])) {
+                        // BAH map drop stat to items_dropped to prevent hash collision in dict keys
```

Comment:
What does `BAH` mean?

Reply:
My initials =) Maybe I should change given we have git annotations

Reply:
I see. Not necessarily crucial to change, maybe I just need to get used to it.